test/core.jl

```diff
@@ -3100,15 +3100,25 @@ Base.convert(::Type{Foo11874},x::Int) = float(x)
 
 # issue #9233
 let
-    err = @test_throws TypeError NTuple{Int, 1}
```

Comment:
It'd be inconsistent with the `@test`. Now everything returns a test object, with nice behaviour at the REPL as a result. I think using `@test_throws` as a one-liner try-catch was not a good idea to begin with, although thats completed unrelated to why I picked the return type.